code/modules/projectiles/guns/projectile/automatic/sts.dm

```diff
@@ -49,44 +51,8 @@
 	. = ..()
 	update_icon()
 
-/obj/item/gun/projectile/automatic/sts/lrifle
-	name = "\"STS\" carbine"
-	desc = "The rugged STS-25, the right left arm of the unfree world. Uses Carbine rounds."
-	icon = 'icons/obj/guns/projectile/sts25.dmi'
-	icon_state = "sts"
-	item_state = "sts"
-	price_tag = 850
-	gun_tags = list(GUN_PROJECTILE,GUN_SILENCABLE, GUN_MAGWELL)
-	matter = list(MATERIAL_PLASTEEL = 20, MATERIAL_WOOD = 12)
-	saw_off = TRUE
-	sawn = /obj/item/gun/projectile/automatic/sts/lrifle/sawn
-
-/obj/item/gun/projectile/automatic/sts/lrifle/update_icon()
-	..()
-
-	var/iconstring = initial(icon_state)
-	var/itemstring = ""
-
-	if (ammo_magazine)
-		iconstring += "[ammo_magazine? "_mag[ammo_magazine.max_ammo]": ""]"
-		itemstring += "_full"
-
-	if (!ammo_magazine || !length(ammo_magazine.stored_ammo))
-		iconstring += "_slide"
-
-	if (silenced)
-		iconstring += "_s"
-		itemstring += "_s"
-
-	icon_state = iconstring
-	set_item_state(itemstring)
-
-/obj/item/gun/projectile/automatic/sts/lrifle/Initialize()
-	. = ..()
-	update_icon()
-
-/obj/item/gun/projectile/automatic/sts/lrifle/sawn
```

Comment:
Keep this as well

Reply:
Isn't it easier to just tell me "just do the opposite, leaving a skeleton STS parent and use "lrifle" for every spawn and recipe?